keras_cv/metrics/coco/pycoco_utils.py

```diff
@@ -0,0 +1,231 @@
+    ]
+
+    gt_annotations = []
+    num_batches = len(groundtruths["source_id"])
+    for i in range(num_batches):
+        max_num_instances = groundtruths["classes"][i].shape[1]
+        batch_size = groundtruths["source_id"][i].shape[0]
+        for j in range(batch_size):
+            num_instances = groundtruths["num_detections"][i][j]
+            if num_instances > max_num_instances:
+                num_instances = max_num_instances
+            for k in range(int(num_instances)):
+                ann = {}
+                ann["image_id"] = groundtruths["source_id"][i][j]
+                ann["iscrowd"] = 0
+                ann["category_id"] = int(groundtruths["classes"][i][j, k])
+                boxes = groundtruths["boxes"][i]
+                ann["bbox"] = [
+                    float(boxes[j, k, 1]),
+                    float(boxes[j, k, 0]),
+                    float(boxes[j, k, 3] - boxes[j, k, 1]),
+                    float(boxes[j, k, 2] - boxes[j, k, 0]),
+                ]
+                ann["area"] = float(
+                    (boxes[j, k, 3] - boxes[j, k, 1])
+                    * (boxes[j, k, 2] - boxes[j, k, 0])
+                )
+                gt_annotations.append(ann)
+
+    for i, ann in enumerate(gt_annotations):
+        ann["id"] = i + 1
+
+    if label_map:
+        gt_categories = [{"id": i, "name": label_map[i]} for i in label_map]
+    else:
+        category_ids = [gt["category_id"] for gt in gt_annotations]
+        gt_categories = [{"id": i} for i in set(category_ids)]
+
+    gt_dataset = {
+        "images": gt_images,
+        "categories": gt_categories,
+        "annotations": copy.deepcopy(gt_annotations),
+    }
+    return gt_dataset
+
+
+def _convert_to_numpy(groundtruths, predictions):
+    """Converts tensors to numpy arrays."""
+    labels = tf.nest.map_structure(lambda x: x.numpy(), groundtruths)
+    numpy_groundtruths = {}
+    for key, val in labels.items():
+        if isinstance(val, tuple):
+            val = np.concatenate(val)
+        numpy_groundtruths[key] = val
+
+    outputs = tf.nest.map_structure(lambda x: x.numpy(), predictions)
+    numpy_predictions = {}
+    for key, val in outputs.items():
+        if isinstance(val, tuple):
+            val = np.concatenate(val)
+        numpy_predictions[key] = val
+
+    return numpy_groundtruths, numpy_predictions
+
+
+def compute_pycoco_metrics(groundtruths, predictions):
+    groundtruths, predictions = _convert_to_numpy(groundtruths, predictions)
+
+    gt_dataset = _convert_groundtruths_to_coco_dataset(groundtruths)
+    coco_gt = COCOWrapper(gt_dataset=gt_dataset)
+    coco_predictions = _convert_predictions_to_coco_annotations(predictions)
+    coco_dt = coco_gt.loadRes(predictions=coco_predictions)
+    image_ids = [ann["image_id"] for ann in coco_predictions]
+
+    coco_eval = cocoeval.COCOeval(coco_gt, coco_dt, iouType="bbox")
+    coco_eval.params.imgIds = image_ids
+    coco_eval.evaluate()
```

Comment:
does this print a ton of logs?

Reply:
Yes it does -- I don't think we can't really do anything about it without forking the Pycocotools code (or with a hack like this: https://stackoverflow.com/questions/8391411/how-to-block-calls-to-print)

Reply:
unfortunate, but if theres no easy solution it should be fine.